plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveInternalRetry.java

```diff
@@ -76,6 +96,16 @@ public synchronized void createDatabase(Database database)
                 super.createDatabase(database);
                 throw new SchemaAlreadyExistsException(database.getDatabaseName());
             }
+
+            @Override
+            public synchronized void dropTable(String databaseName, String tableName, boolean deleteData)
+            {
+                // Simulate retry mechanism with timeout failure of ThriftHiveMetastore.
+                // 1. dropTable correctly drops table but timeout is triggered
+                // 2. Retry to dropTable throws TableNotFoundException
+                super.dropTable(databaseName, tableName, deleteData);
+                throw new TableNotFoundException(schemaTableName(databaseName, tableName));
```

Comment:
same as for Iceberg test, we should do this only for  tables named `test_drop_internal_retry`

Reply:
I would leave it. Whole test class is about testing retry, so simulate timeout for all calls feels ok for me.